src/sqlfluff/utils/reflow/elements.py

```diff
@@ -662,7 +671,12 @@ def respace_point(
             if last_whitespace:
                 ws_idx = self.segments.index(last_whitespace)
                 if ws_idx > 0:
-                    prev_seg = self.segments[ws_idx - 1]
+                    for prev_seg in self.segments[ws_idx - 1 :: -1]:
+                        # Skip past any indents
+                        if prev_seg.is_type("indent"):
+                            continue
```

Comment:
Nit: The `continue` / `break` together reads a little strange. No big deal, though.